ts/packages/aiclient/src/openai.ts

```diff
@@ -506,23 +514,37 @@ function createAzureOpenAIChatModel(
         if (!result.success) {
             return result;
         }
+
+        let fullResponseText = "";
+        let tokenUsage;
         return {
             success: true,
             data: (async function* () {
                 for await (const evt of readServerEventStream(result.data)) {
                     if (evt.data === "[DONE]") {
+                        try {
```

Comment:
Do you want this to be always logged?  This can be very noisy especially when running tests?  In general, telemetry should be more targeted, no?

Reply:
We could put this behind a setting/telemetry logging level and add an @ command so it can be configured per user?

Reply:
In the case that triggered the addition of the feature we would like it always logged, but I totally agree that having a setting controlling this is a good idea. I would even make it opt-in once we add it. Can you point me to an existing example for adding a setting and associated command? I can then make it so we can enable/disable at the app level, in addition to letting the user override it.

Reply:
Take a look at **@config dev on** implemented in: https://github.com/microsoft/TypeAgent/blob/main/ts/packages/dispatcher/src/handlers/configCommandHandlers.ts

Reply:
And don't get distracted by the config command handler since that does a lot of stuff.  For a simple command, check out this guy: https://github.com/microsoft/TypeAgent/blob/main/ts/packages/dispatcher/src/handlers/tokenCommandHandler.ts

Reply:
Also note that this library `aiclient` isn't only used in the `shell` or `cli interactive`, it is used in many other cases including examples and test generation, etc.   I assume that you want only log a certain use case, which mean probably a parameter to pass in a logger might be a better choice here? (as well as the switch to turn it on and off).

Reply:
> Also note that this library `aiclient` isn't only used in the `shell` or `cli interactive`, it is used in many other cases including examples and test generation, etc. I assume that you want only log a certain use case, which mean probably a parameter to pass in a logger might be a better choice here? (as well as the switch to turn it on and off).

Very good point...I do like the "pass in the logger" as a parameter.  If it's not there we don't have to do anything with it... +1

Reply:
I just pushed a change adding a check for an environment variable that controls whether logging should happen or not, keeping it disabled by default. 

In the end I went with this as a first step because all kinds of requests would be useful for my needs, but if you feel strongly that I should instead condition it only to certain types of model consumption I can move the check for the environment variable to those entry points where the models are created and pass the enable flag as a parameter from there.

While making this change, I also noticed the tags parameter in the CompleteChat requests could be a useful value to add to the events, so I added it. This should help in cases we need to filter out prompt data logging events by area, like it can be done with token counting.